backend/Social/SocialAPI.py

```diff
@@ -45,11 +49,63 @@ def profiles():
         '''
         if request.method == 'PUT':
             data = request.get_json()
-            if "profilePictureURI" in data:
-                data["profilePictureUrl"] = data.pop("profilePictureURI")
-
-            result = db.Profiles.update_one(
-                {"userID": data["userID"]}, {'$set': data}, upsert=True)
+            imgString = str(data["image_uri"]) # get image URI/URL/base64 from JSON request
+            userID = data["userID"] # get userID from JSON request
+            # START OF PROFILE PICTURE GENERATION CODE
+            if imgString[:5] == "data:":   # given image string is in URI form         
+                imgType = DataURI(imgString).mimetype # get FILE_FORMAT (mimetype) of image URI
+                imgBinary = DataURI(imgString).data
+                imgFileName = "profile_pic." + str(imgType)[str(imgType).find("/")+1:] # name image to be generated as "profile_pic.<FILE_FORMAT>"
+                fd = open(imgFileName, 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[:4] == "/9j/": # given image string is in base64 form and in JPG/JPEG format
+                imgBinary = a2b_base64(imgString)
+                fd = open('profile_pic.jpg', 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[:6] == "iVBORw": # given image string is in base64 form and in PNG format
+                imgBinary = a2b_base64(imgString)
+                fd = open('profile_pic.png', 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[:6] == "R0lGOD": # given image string is in base64 form and in GIF format
+                imgBinary = a2b_base64(imgString)
+                fd = open('profile_pic.gif', 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[0] == "U": # given image string is in base64 form and in WEBP format
+                imgBinary = a2b_base64(imgString)
+                fd = open('profile_pic.webp', 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[:4] == "SUkq": # given image string is in base64 form and in TIFF format
+                imgBinary = a2b_base64(imgString)
+                fd = open('profile_pic.tiff', 'wb')
+                fd.write(imgBinary)
+                fd.close()
+            elif imgString[:4] == "http": # given image string is in URL form (assuming all images stored online are in JPEG format)
+                response = urllib.request.urlopen(imgString)
+                fd = open('profile_pic.jpg','wb')
+                fd.write(response.read())
+                fd.close()
+            # END OF PROFILE PICTURE GENERATION CODE    
+            imgFileLocation = os.getcwd() # get location of generated image file
+            imgKey = str(userID) + "/profile_pic." + str(imgType)[str(imgType).find("/")+1:] # name of ImageKey for generated image file
```

Comment:
You are using imgType here for all possible cases, but its not initalised in this scope and only case where URI string is data: assigns information to the variable imgType.